plugins/alembic/module/vtkF3DAlembicReader.cxx

```diff
@@ -20,51 +22,338 @@
+    if (needToDuplicate)
+    {
+      auto faceCount = originalData.Indices.size();
+      duplicatedData.Indices.resize(faceCount);
+      for (size_t i = 0; i < faceCount; i++)
+      {
+        auto thisFaceVertexCount = originalData.Indices[i].size();
+
+        // TODO check with nYue
+        duplicatedData.Indices[i].resize(
+          thisFaceVertexCount, Alembic::Abc::V3i(-9999, -9999, -9999));
+      }
+
+      // Points
+      {
+        V3fContainer pV3F;
+        int pRunningIndex = 0;
+        for (size_t i = 0; i < faceCount; i++)
+        {
+          auto thisFaceVertexCount = originalData.Indices[i].size();
+          for (size_t j = 0; j < thisFaceVertexCount; j++)
+          {
+            Alembic::Abc::V3f original_position =
+              originalData.Attributes.at("P")[originalData.Indices[i][j].x];
+            pV3F.emplace_back(original_position);
+            duplicatedData.Indices[i][j].x = pRunningIndex;
+            pRunningIndex++;
+          }
+        }
+
+        duplicatedData.Attributes.insert(AttributesContainer::value_type("P", pV3F));
+      }
+
+      // UV
+      if (haveUV)
+      {
+        V3fContainer uvV3F;
+        int uvRunningIndex = 0;
+
+        for (size_t i = 0; i < faceCount; i++)
+        {
+          auto thisFaceVertexCount = originalData.Indices[i].size();
+          for (size_t j = 0; j < thisFaceVertexCount; j++)
+          {
+            Alembic::Abc::V3f originalUV =
+              originalData.Attributes.at("uv")[originalData.Indices[i][j].y];
+            uvV3F.emplace_back(originalUV);
+            duplicatedData.Indices[i][j].y = uvRunningIndex;
+            uvRunningIndex++;
+          }
+        }
+
+        duplicatedData.Attributes.insert(AttributesContainer::value_type("uv", uvV3F));
+      }
+
+      // Normal
+      if (haveN)
+      {
+        V3fContainer nV3F;
+        int nRunningIndex = 0;
+
+        for (size_t i = 0; i < faceCount; i++)
+        {
+          auto thisFaceVertexCount = originalData.Indices[i].size();
+          for (size_t j = 0; j < thisFaceVertexCount; j++)
+          {
+            Alembic::Abc::V3f original_N =
+              originalData.Attributes.at("N")[originalData.Indices[i][j].z];
+            nV3F.emplace_back(original_N);
+            duplicatedData.Indices[i][j].z = nRunningIndex;
+            nRunningIndex++;
+          }
+        }
+
+        duplicatedData.Attributes.insert(AttributesContainer::value_type("N", nV3F));
+      }
+    }
+    else
+    {
+      duplicatedData = originalData;
+    }
+  }
+
+  void FillPolyData(const IntermediateGeometry& data, vtkPolyData* polydata)
+  {
+    // Create 10 points.
+    vtkNew<vtkPoints> points;
+    vtkNew<vtkCellArray> cells;
+
+    auto pMapIter = data.Attributes.find("P");
+    if (pMapIter == data.Attributes.end())
+    {
+      // TODO throw an error instead and catch it above ?
+      vtkWarningWithObjectMacro(nullptr, "A data could not be read, leaving empty");
+      return;
+    }
```

Comment:
A warning here seems fine, wdyt @nyue ?

Reply:
Having no "P" is still a valid Alembic though it makes no sense in a geometry viewer. For example, Alembic camera will have no "P" which is a valid use case but for f3d not really (unless there are plans to support camera)

Reply:
Then we should not warn at all.

> Unless there are plans to support camera

We do support reading cameras but only for importers. FYI @Meakk

Reply:
could you share such a file, containing camera I mean @nyue ?

Reply:
[alembic_camera.zip](https://github.com/f3d-app/f3d/files/13401092/alembic_camera.zip)

Reply:
This file does not reach this code because there is neither `Alembic::AbcGeom::IXform::matches` nor `Alembic::AbcGeom::IPolyMesh::matches`.

So maybe this code is unreachable after all.

Reply:
The alembic_camera.zip contains an alembic file purely with camera information, no geometry at all i.e. no "P", no polymesh.

Reply:
Let's drop camera support for now.

Reply:
Of course, we have no way to support camera. I'm just trying to improve coverage here.